--- a/sqlcipher/sqlite3.c
+++ b/sqlcipher/sqlite3.c
@@ -114904,7 +114904,8 @@
   db->init.iDb = targetDb_idx;
 
   db->flags |= SQLITE_WriteSchema | SQLITE_IgnoreChecks;
-  db->mDbFlags |= DBFLAG_PreferBuiltin | DBFLAG_Vacuum;
+  /* Shep: preserve implicit rowids in unindexed tables during export too. */
+  db->mDbFlags |= DBFLAG_PreferBuiltin | DBFLAG_Vacuum | DBFLAG_VacuumInto;
   db->flags &= ~(u64)(SQLITE_ForeignKeys | SQLITE_ReverseOrder | SQLITE_Defensive | SQLITE_CountRows);
   db->mTrace = 0;
 
